SECURITY.md

```diff
@@ -1,8 +1,133 @@
+issues or social posts) until we have addressed it and granted
+permission to disclose.
+We may coordinate with you on a public disclosure timeline.
+
+By reporting to us, you agree to **participate in coordinated
+vulnerability disclosure**, allowing us time to develop and
+distribute a fix before details are made public.
+
+------------------------------------------------------------------------
+
+## Bug Bounty Program Overview
+
+Our bug bounty program rewards valid security bug reports submitted
+through **HackerOne**, where bounty amounts depend on severity and
+impact.
+
+**Scope:** Core Cosmos Stack components, such as: Cosmos SDK, CometBFT, IBC, Cosmos EVM and
+other critical infrastructure
+
+For full scope, severity definitions, and reward ranges, see the Cosmos
+page on [HackerOne](https://hackerone.com/cosmos).
+
+We follow a **Safe Harbor** policy to protect researchers acting in good
+faith. The official HackerOne page contains the **Coordinated
+Vulnerability Disclosure Policy** and **Safe Harbor terms**.
+> In case of discrepancies, the HackerOne policy supersedes other
+documents.
+
+------------------------------------------------------------------------
+
+## Vulnerability Severity Levels
+
+All reported vulnerabilities are assigned a severity category guiding how they are handled and disclosed.
+
+| **Level**    | **Description**                                                                     | **Examples**                                            |
+|--------------|-------------------------------------------------------------------------------------|---------------------------------------------------------|
+| **Critical** | Bugs posing an existential or network-wide threat (fund loss, inflation, or theft). | Token creation beyond fixed supply, permanent fork bug. |
+| **High**     | Major disruption to many nodes or users, often remotely exploitable.                | Remote crash or chain halt vulnerability.               |
+| **Medium**   | Moderate impact or harder to exploit; may require specific configurations.          | Node halt requiring high permissions and access         |
+| **Low**      | Minor impact or impractical exploitation.                                           | Slow block propagation, limited DoS.                    |
+
+These levels align with common security practices and determine both
+urgency and disclosure timelines. For more information, refer to our [Classification Matrix](./resources/CLASSIFICATION_MATRIX.md).
+
+------------------------------------------------------------------------
+
+## Silent Patch and Disclosure Process
+
+Cosmos adopts a **silent patch** approach. Vulnerabilities are fixed
+quietly before disclosure.
+This mirrors practices from projects like **Ethereum's Geth**, **Bitcoin
+Core**, and **Zcash**.
+
+Announcing vulnerabilities too early can endanger unpatched nodes.
+Silent fixes allow time for safe upgrades before attackers become aware.
```

Comment:
True, I'm curious if we can stick to this if we see a chain get attacked (and let's say get halted). So this doc should mention what we do in case we know that attackers have already become aware before a maintenance release. My guess would be that we treat is as critical, although it is not critical.